### PR TITLE
Adding Nice DCV session workflows to HPC

### PR DESCRIPTION
**Pull request by alijawadfahs**

### CODE REVIEW

**Review by marounkoussaifi — APPROVED** [review]

**Review by mboussaa — APPROVED** [review]
LGTM, I gave some feedback by slack to Ali

File: HPC/resources/catalog/Create_Nice_DCV_Node.xml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    <variable name="AWS_Key" value="" model="PA:HIDDEN" description="" group="" advanced="false" hidden="false"/>
+    <variable name="AWS_Secret" value="" model="PA:HIDDEN" description="" group="" advanced="false" hidden="false"/>
+    <variable name="SSH_Key" value="" model="" description="" group="" advanced="false" hidden="false"/>
+    <variable name="Instance_type" value="t2.large" model="PA:NOT_EMPTY_STRING"   advanced="false" hidden="false"/>
+    <variable name="AMI_Id" value="ami-0081c1d642678a3ff" model="PA:NOT_EMPTY_STRING"   advanced="false" hidden="false"/>
+    <variable name="SubNetId" value="subnet-xxx" model="PA:NOT_EMPTY_STRING" description="" group="" advanced="false" hidden="false"/>
+    <variable name="SecurityGroup" value="sg-xxx" model="PA:NOT_EMPTY_STRING" description="" group="" advanced="false" hidden="false"/>
+    <variable name="Region" value="eu-west-1" model="PA:NOT_EMPTY_STRING"   advanced="false" hidden="false"/>
+    <variable name="VpcId" value="vpc-xxx" model="PA:NOT_EMPTY_STRING" description="" group="" advanced="false" hidden="false"/>
+  </variables>
```
Comment (aminelouati) [inline HPC/resources/catalog/Create_Nice_DCV_Node.xml:20]: As you have a high number of variables, could you organize them similarly to the AWSEC2Infrastructure using the group attribute ?

File: HPC/resources/catalog/Create_Nice_DCV_Node.xml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.13" xsi:schemaLocation="urn:proactive:jobdescriptor:3.13 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.13/schedulerjob.xsd"  name="Create_Nice_DCV_Node" projectName="AWS Nice DCV" priority="normal" onTaskError="continueJobExecution"  maxNumberOfExecution="2"  >
+  <variables>
+    <variable name="NS_name" value="aws-dcv" model="PA:NOT_EMPTY_STRING"   advanced="false" hidden="false"/>
+    <variable name="rm_host_name" value="try.activeeon.com" model="PA:NOT_EMPTY_STRING" description="" group="" advanced="false" hidden="false"/>
+    <variable name="tokens" value="DCV"    advanced="false" hidden="false"/>
+    <variable name="MAX_VMs" value="1" model="PA:INTEGER"   advanced="false" hidden="false"/>
+    <variable name="maxNodePerVM" value="1" model="PA:INTEGER"   advanced="false" hidden="false"/>
+    <variable name="AWS_Key" value="" model="PA:HIDDEN" description="" group="" advanced="false" hidden="false"/>
+    <variable name="AWS_Secret" value="" model="PA:HIDDEN" description="" group="" advanced="false" hidden="false"/>
```
Comment (aminelouati) [inline HPC/resources/catalog/Create_Nice_DCV_Node.xml:12]: For AWS_Key and AWS_Secret I think that is better provide them as follow
<variable name="AWS_Key" value=""     />
 <variable name="AWS_Secret" value="$AWS_Key" model="PA:CREDENTIAL"    />
 The hidden model it is not safe

File: HPC/resources/catalog/Create_Nice_DCV_Node.xml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                                awsSecret,//awsSecretKey
+                               	maxVms,//maxVms
+                               	InstanceType,//defaultInstanceType
+                               	amiId,//amiId
+                               	publicSshKey,//publicSshKey
+                            	VpcId,//defaultVpcId
+                                defaultSubNetId,//defaultSubNetId
+                                defaultSecurityGroup,//defaultSecurityGroup
+                                region,//region
+                                "pamr://4096/",//rmUrl
```
Comment (aminelouati) [inline HPC/resources/catalog/Create_Nice_DCV_Node.xml:91]: please add a comment here about the use of 4096 which is very specific to our try platforms. Normally it should be 0

File: HPC/resources/catalog/Connect_Nice_DCV.xml [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+  </variables>
+  <description>
+    <![CDATA[ A workflow that creates a virtual Nice DCV session and report back the endpoint and the username that can be used to connect to the session. When a stop signal is transmitted the workflow will end the session and delete the associated Linux user. ]]>
+  </description>
+  <genericInformation>
+    <info name="NODE_ACCESS_TOKEN" value="DCV"/>
+    <info name="bucketName" value="controls"/>
+    <info name="workflow.icon" value="/automation-dashboard/styles/patterns/img/wf-icons/DCV.png"/>
+    <info name="Documentation" value="user/ProActiveUserGuide.html#_task_signal_api"/>
+    <info name="group" value="scheduleradmins"/>
```
Comment (aminelouati) [inline HPC/resources/catalog/Connect_Nice_DCV.xml:19]: why the user group is scheduleradmins ?

File: HPC/resources/catalog/Create_Nice_DCV_Node.xml (hunk 1 above)
Comment (alijawadfahs) [reply HPC/resources/catalog/Create_Nice_DCV_Node.xml:20]: Done

File: HPC/resources/catalog/Create_Nice_DCV_Node.xml (hunk 2 above)
Comment (alijawadfahs) [reply HPC/resources/catalog/Create_Nice_DCV_Node.xml]: Done

File: HPC/resources/catalog/Create_Nice_DCV_Node.xml (hunk 3 above)
Comment (alijawadfahs) [reply HPC/resources/catalog/Create_Nice_DCV_Node.xml]: Done

File: HPC/resources/catalog/Connect_Nice_DCV.xml (hunk 4 above)
Comment (alijawadfahs) [reply HPC/resources/catalog/Connect_Nice_DCV.xml]: changed to "public-objects"

**Review by aminelouati — APPROVED** [review]

**Pull request merged**